### ISSUE TITLE
Add watch function to KV API

### ISSUE THREAD

**Issue by autodidaddict** [mmmries/jetstream#71]
The `contents` function is basically a watch function that times out after 100ms. I'd like the ability to request a long-lived consumer of key-value pairs so that every time a new key is added (or deleted), my consumer gets notified of this.

My initial attempt was to write a watch function and then modify the `contents` function to basically be a specialized consumer of watch. However, I quickly ended up unsure on how to proceed.

What I was trying to implement:

```elixir
def watch(conn, bucket_name, module, keys \\ "*")
```

Where `module` was a module that uses `GnatServer`.. but what I really want is to receive a message like this:

```elixir
{:key_added, key, value}
{:key_removed, key, value}
{:key_updated, key, value}
```
So that felt more like something that made more sense with a `Gnat.sub` .. but again I wasn't sure how to translate the stream into these three pieces because the documentation on the headers used for KV is sparse to say the least.

**Comment by autodidaddict** [comment]
I'm more than happy to contribute this, I think I just want some directional advice on what you think is the best approach here.

**Comment by autodidaddict** [comment]
here's some as-yet-untested code (brain-compiled) that describes what I was thinking:

```elixir
def watch(conn, bucket_name, handler) do
    stream = stream_name(bucket_name)
    inbox = Util.reply_inbox()
    consumer_name = "all_key_values_watcher_#{Util.nuid()}"

    {:ok, int_handler} = Task.start(fn ->
      receive do
        {:msg, %{topic: key, body: body, headers: headers}} ->
          key = subject_to_key(key, bucket_name)

          if {"kv-operation", "DEL"} in headers do
            handler.(:key_deleted, key, body)
          else if {"kv-operation", "PUT"} in headers  # ??? PUT ?
            handler.(:key_added, key, body)
          else
            handler.(:key_updated, key, body)
          end
      end
    end)

    with {:ok, sub} <- Gnat.sub(conn, int_handler, inbox),
         {:ok, _consumer} <-
           Consumer.create(conn, %Consumer{
             durable_name: consumer_name,
             deliver_subject: inbox,
             stream_name: stream,
             ack_policy: :none,
             max_ack_pending: -1,
             max_deliver: 1
           }) do

        {:ok, sub}
    end
  end
```

And then it would be up to the caller of `watch` to decide when to unsubscribe and remove the consumer (probably through `unwatch` that takes the bucket name and the `sub` pid?)

**Comment by brandynbennett** [comment]
Released as 0.0.6
